cms/server/contest/handlers/contest.py

```diff
@@ -312,27 +312,30 @@ def _get_current_user_from_cookie(self):
     def setup_locale(self):
         lang_codes = list(iterkeys(self.available_translations))
 
-        # Select the one the user likes most.
-        basic_lang = 'en'
-        if basic_lang not in lang_codes:
-            basic_lang = lang_codes[0].replace("_", "-")
-
-        http_langs = [lang_code.replace("_", "-") for lang_code in lang_codes]
-        self.browser_lang = parse_accept_header(
+        browser_langs = parse_accept_header(
             self.request.headers.get("Accept-Language", ""),
-            LanguageAccept).best_match(http_langs, basic_lang)
-
-        self.cookie_lang = self.get_cookie("language", None)
-
-        if self.cookie_lang in http_langs:
-            lang_code = self.cookie_lang
+            LanguageAccept).values()
```

Comment:
Does it make sense to cache this? parse_accept_header is a pretty complex regex...